query-engine/driver-adapters/src/conversion/js_to_quaint.rs

```diff
@@ -218,7 +218,9 @@ pub fn js_value_to_quaint(
                 // DbNull
                 serde_json::Value::Null => Ok(QuaintValue::null_json()),
                 // JsonNull
-                serde_json::Value::String(s) if s == "$__prisma_null" => Ok(QuaintValue::json(serde_json::Value::Null)),
+                serde_json::Value::String(s) => serde_json::from_str(&s)
+                    .map_err(|_| conversion_error!("Failed to parse json"))
```

Comment:
```suggestion
                    .map_err(|_| conversion_error!("Failed to parse incoming json from a driver adapter"))
```

It should never really happen but for the case it will for whatever reason - a bit more context.